modules/ui.go

```diff
@@ -0,0 +1,207 @@
+package modules
+
+import (
+	"embed"
+	"strings"
+
+	"github.com/GoMudEngine/GoMud/internal/events"
+	"github.com/GoMudEngine/GoMud/internal/plugins"
+	"github.com/GoMudEngine/GoMud/internal/rooms"
+	"github.com/GoMudEngine/GoMud/internal/users"
+	"github.com/GoMudEngine/GoMud/internal/mudlog"
+)
+
+var (
+	//go:embed ui/*
+	ui_Files embed.FS
+	
+	// Instance accessible from elsewhere in this package
+	uiModule UIModule
+)
+
+func init() {
+	// Create the module instance
+	uiModule = UIModule{
+		plug: plugins.New(`ui`, `1.0`),
+	}
+
+	// Attach the file system to the plugin
+	if err := uiModule.plug.AttachFileSystem(ui_Files); err != nil {
+		panic(err)
+	}
+	
+	// Set default values
+	uiModule.setDefaultValues()
+	
+	// Register the onLoad callback to load config values after plugin system is ready
+	uiModule.plug.Callbacks.SetOnLoad(func() {
+		uiModule.RefreshConfig()
+	})
+
+	// Register the user commands
+	uiModule.plug.AddUserCommand(`gui`, GUICommand, true, false)
+	uiModule.plug.AddUserCommand(`checkgui`, CheckGUICommand, true, false)
+}
+
+// UIModule stores configuration and state for the UI module
+type UIModule struct {
+	plug        *plugins.Plugin
+	uiUrl       string
+	uiVersion   string
+	removeField string // Field name for removal flag
+}
+
+// Set hardcoded default values for initialization
+func (g *UIModule) setDefaultValues() {
+	g.uiUrl = "https://github.com/MorquinDevlar/GoMudUI/releases/latest/download/gomud-ui.mpackage"
+	g.uiVersion = "1.0.0"
+	g.removeField = "remove"
```

Comment:
Same comment as before - these should come from the config file - any values set in the `data-overlays/config.yaml` would de-facto be the default (it's read-only). Only when these values are changed in the overrides file, or via the `server set` command do they change.